### PR TITLE
Support gRPC health checking

### PR DESCRIPTION
**Pull request by honnix**
# TL;DR
Support gRPC health checking.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
The implementation of https://github.com/grpc/grpc/blob/master/doc/health-checking.md is part of go gRPC lib.

## Tracking Issue
https://github.com/lyft/flyte/issues/489

## Follow-up issue
NA

### CODE REVIEW

**Comment by honnix** [comment]
@kumare3 Would like to take a look at this as well? Thanks.

File: cmd/entrypoints/serve.go [hunk 1]
```diff
@@ -54,6 +56,11 @@ func serveInsecure(ctx context.Context, cfg *config.Config) error {
 func newGRPCServer(_ context.Context, cfg *config.Config) *grpc.Server {
 	grpcServer := grpc.NewServer()
 	datacatalog.RegisterDataCatalogServer(grpcServer, datacatalogservice.NewDataCatalogService())
+
+	healthServer := health.NewServer()
+	healthServer.SetServingStatus("", grpc_health_v1.HealthCheckResponse_SERVING)
+	grpc_health_v1.RegisterHealthServer(grpcServer, healthServer)
```
Comment (chanadian) [inline cmd/entrypoints/serve.go:62]: 👍 nice
nit: we can rename the http health check method below to `serveHTTPHealthcheck`, this is for integration into proxies like Envoy's health check

**Review by chanadian — DISMISSED** [review]

File: cmd/entrypoints/serve.go (hunk 1 above)
Comment (honnix) [reply cmd/entrypoints/serve.go:62]: PTAL again and help merge it. Thanks.

**Comment by chanadian** [comment]
Looks good to me @honnix , @kumare3 would you like to coordinate merging and getting this out since your a deploying on cadence?

**Comment by kumare3** [comment]
LGTM, Merging

**Pull request merged**